packages/eslint-plugin/src/rules/strict-void-return.ts

```diff
@@ -0,0 +1,991 @@
+    ): void {
+      if (
+        propNode.value.type === AST_NODE_TYPES.AssignmentPattern ||
+        propNode.value.type === AST_NODE_TYPES.TSEmptyBodyFunctionExpression
+      ) {
+        return;
+      }
+      const propName = sourceCode.getText(propNode.key);
+      const propTsNode = parserServices.esTreeNodeToTSNodeMap.get(propNode);
+
+      if (propTsNode.kind === ts.SyntaxKind.MethodDeclaration) {
+        // Object property is a method shorthand.
+
+        if (propTsNode.name.kind === ts.SyntaxKind.ComputedPropertyName) {
+          // Don't check object methods with computed name.
+          return;
+        }
+        const objTsNode = propTsNode.parent;
+        assert(ts.isObjectLiteralExpression(objTsNode));
+        const objType = checker.getContextualType(objTsNode);
+        if (objType == null) {
+          // Expected object type is unknown.
+          return;
+        }
+        const propSymbol = checker.getPropertyOfType(
+          objType,
+          propTsNode.name.text,
+        );
+        if (propSymbol == null) {
+          // Expected object type is known, but it doesn't have this method.
+          return;
+        }
+        const propExpectedType = checker.getTypeOfSymbolAtLocation(
+          propSymbol,
+          propTsNode,
+        );
+        if (isVoidReturningFunctionType(propExpectedType)) {
+          reportIfNonVoidFunction(propNode.value, 'Prop', { propName });
+        }
+        return;
+      }
+
+      // Object property is a regular property.
+      checkExpressionNode(propNode.value, 'Prop', { propName });
+    }
+
+    /**
+     * Finds errors in a class property.
+     *
+     * In addition to the regular check against the contextual type,
+     * we also check against the base class property (when the class extends another class)
+     * and the implemented interfaces (when the class implements an interface).
+     *
+     * This can produce multiple errors at once.
+     */
+    function checkClassPropertyNode(
+      propNode: TSESTree.PropertyDefinition,
+    ): void {
+      if (propNode.value == null) {
+        return;
+      }
+      const propTsNode = parserServices.esTreeNodeToTSNodeMap.get(propNode);
+      const memberName = sourceCode.getText(propNode.key);
+      const className = propNode.parent.parent.id?.name ?? 'class';
+
+      // Check in comparison to the base class property.
+      if (options.considerBaseClass) {
+        for (const {
+          baseType,
+          baseMemberType,
+        } of util.getBaseTypesOfClassMember(
+          checker,
+          propTsNode,
+          ts.SyntaxKind.ExtendsKeyword,
+        )) {
+          const baseName = baseType.getSymbol()?.name ?? 'class';
+          if (isVoidReturningFunctionType(baseMemberType)) {
+            reportIfNonVoidFunction(propNode.value, 'ExtMember', {
+              memberName,
+              className,
+              baseName,
+            });
+          }
```

Comment:
[Bug] It's kind of unusual to report twice in one place. Plus, for better performance, it's nice to not have to do checks later on once there's already a known report. For here and elsewhere, let's skip later reports as soon as one is made:

```suggestion
            });
            return;
          }
```